LykkeApi2/Controllers/HistoryController.cs

```diff
@@ -93,57 +60,67 @@ public IActionResult RequestClientHistoryCsv([FromBody]RequestClientHistoryCsvRe
         {
-            return Ok(new GetClientHistoryCsvResponseModel { Url = await _historyExportsRepository.GetUrl(_requestContext.ClientId, id)});
+            return Ok(new GetClientHistoryCsvResponseModel { Url = await _historyExportsRepository.GetUrl(_requestContext.ClientId, id) });
         }
 
         /// <summary>
         /// Getting history by wallet identifier
         /// </summary>
         /// <param name="walletId">Wallet identifier</param>
-        /// <param name="operationType">The type of the operation, possible values: CashIn, CashOut, Trade, LimitTrade, LimitTradeEvent</param>
+        /// <param name="operationType">The type of the operation, possible values: CashIn, CashOut, Trade, OrderEvent</param>
         /// <param name="assetId">Asset identifier</param>
         /// <param name="assetPairId">Asset pair identifier</param>
         /// <param name="take">How many maximum items have to be returned</param>
         /// <param name="skip">How many items skip before returning</param>
         /// <returns></returns>
         [HttpGet("wallet/{walletId}")]
         [SwaggerOperation("GetByWalletId")]
-        [ProducesResponseType(typeof(ErrorResponse), (int) HttpStatusCode.InternalServerError)]
-        [ProducesResponseType(typeof(void), (int) HttpStatusCode.NotFound)]
-        [ProducesResponseType(typeof(IEnumerable<HistoryResponseModel>), (int) HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType(typeof(void), (int)HttpStatusCode.NotFound)]
+        [ProducesResponseType(typeof(IEnumerable<HistoryResponseModel>), (int)HttpStatusCode.OK)]
         public async Task<IActionResult> GetByWalletId(
             string walletId,
-            [FromQuery(Name = "operationType")] HistoryOperationType[] operationType,
+            [FromQuery(Name = "operationType")] string[] operationType,
             [FromQuery] string assetId,
             [FromQuery] string assetPairId,
             [FromQuery] int take,
             [FromQuery] int skip)
         {
             var clientId = _requestContext.ClientId;
 
-            var wallets = await _clientAccountService.GetWalletsByClientIdAsync(clientId);
-
-            if (!wallets.Any(x => x.Id.Equals(walletId)))
+            // TODO: should be removed after release. operationType parameter should be of type HistoryType[]
+            var types = new HashSet<HistoryType>();
+            foreach (var opType in operationType)
             {
-                return NotFound();
+                if (Enum.TryParse<HistoryType>(opType, out var result))
+                    types.Add(result);
             }
 
-            var response = await _operationsHistoryClient.GetByWalletId(walletId, operationType, assetId, assetPairId, take, skip);
+            var wallets = (await _clientAccountService.GetWalletsByClientIdAsync(clientId)).ToList();
```

Comment:
Code 107-118 can be simplified into:
var wallet = clientAccountService.GetWalletAsync(waletId));
if(wallet == null)
  return NotFound();
if(wallet.Type == "Trading")
  walletId = clientId;